src/foundry/client/pixi/perception/detection-mode.d.ts

```diff
@@ -0,0 +1,199 @@
+export {};
+
+// TODO: Move to common/documents/token
+type TokenDetectionMode = {
+  /** The id of the detection mode, a key from CONFIG.Canvas.detectionModes */
+  id: string;
+  /** Whether or not this detection mode is presently enabled */
+  enabled: boolean;
+  /** The maximum range in distance units at which this mode can detect targets */
+  range: number;
+};
+// TODO: Move to client/pixi/layers/effects/visibility.js
+type CanvasVisibilityTest = {
+  point: PIXI.Point;
+  los: Map<VisionSource, boolean>;
+};
+type CanvasVisibilityTestConfig = {
+  object: PlaceableObject;
+  tests: CanvasVisibilityTest[];
+};
+
+// TODO: Remove after foundry.abstract.DataModel is defined
+// Currently that is in PR #2331 (branch v10/non-inferring-data-fields)
```

Comment:
I mentioned wanting all TODOs solved before canvas-updates is merged. This would be the exception.